src/model/GroundHeatExchangerVertical_Impl.hpp

```diff
@@ -56,6 +56,8 @@ namespace model {
 
       boost::optional<int> numberofBoreHoles() const;
 
+      boost::optional<double> boreHoleTopDepth() const;
```

Comment:
Make it required (double)

Reply:
Making it consistent would at least be IDD having a default of 1, getter being still double not optional but getting the default, and ctor certainly not hardcoding it to 1 (and you wouldn't need VT)
I'd still prefer required field, no resetter, less api area to maintain. The alternative is easier just because you don't need VT, but you already did it anyways

Reply:
**any field should have default or required-field (prefer required-field)** unless there is no default or reasonable value, or eplus uses it based on another choice field, or blank means turn off something or something weird like that.

Reply:
I'll follow the path of making it required-field, but I don't understand how my previous implementation wasn't consistent with the other fields. Take bore hole length for example -- it has no default in the idd, and the ctor sets it to 76.2...

Reply:
Ok fair, you were consistent, I hadn't checked the existing implementation in details and I should have said "at least correct".

The existing implementation is awful for BoreholeLength.

* Field is neither \required-field nor has a \default, which isn't justified and wrong
* Having something optional, but hardcode in Ctor is wrong
* It has a resetter, so it can be empty, FT only fills it it's set, but Borehole Length is a \required-field in E+ IDD...